Sources/SwiftLexicalLookup/SimpleLookupQueries.swift

```diff
@@ -0,0 +1,176 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+
+extension SyntaxProtocol {
+  /// Given syntax node position, returns all available labeled statements.
+  @_spi(Compiler) @_spi(Testing) public func lookupLabeledStmts() -> [LabeledStmtSyntax] {
+    return lookupLabeledStmts(at: self)
+  }
+
+  /// Given syntax node position, returns the current switch case and it's fallthrough destination.
+  @_spi(Compiler) @_spi(Testing) public func lookupFallthroughSourceAndDest()
```

Comment:
We can give this an `@_spi` because we're still working on it, but I would like us to treat it like it's going to become real API (and not just for the compiler). To that end, I think we should replace the `@_spi(Compiler) @_spi(Testing)` on these with one `@_spi(Experimental)` that we can remove later.

Additionally, let's document these entry points a bit more thoroughly here, so that when we drop the "experimental" we won't have to come back and improve things. For example, the goal of this function is to tell us, for a `fallthrough` statement, the case that it's coming from and the case it's going to. The documentation should provide a short code snippet showing the kind of code it's expected to be used on, e.g.,

```swift
switch value {
  case 2:
    doSomething()
    fallthrough
  case 1:
    doSomethingElse()
  default:
    break
}
```

When calling on that `fallthrough`, we'll get back the syntax nodes for `case 2` and `case 1`, in that order. The `nil` results are there to handle ill-formed code: there's no `source` if the fallthrough is outside of a case. There's no `destination` if there is no case or default after the `source`. We'll help our future users considerably if we write this out, and also help prove to ourselves that the API is doing what we need.

Which brings up a question: should this API only be available on `FallthroughStmtSyntax` rather than on `SyntaxProtocol`? It's a bit unlike the other two queries in that it's really for exactly one statement node.

Reply:
That’s a good point. I added documentation to the three methods. I hope it's sufficient sufficient. I also moved `lookupFallthroughSourceAndDest` to `FallthroughStmtSyntax`. I also think it makes more sense, since it’s quite specific for that one particular node.

Reply:
All of these changes look great!